src/main/java/core/basesyntax/Circle.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax;
+
+public class Circle extends Figure {
+    private double radius;
+
+    public Circle() {
+    }
+
+    public Circle(double radius) {
+        this.typeOfFigure = TypeOfFigure.CIRCLE;
+        this.area = getCircleArea(radius);
```

Comment:
Avoid calling methods inside the constructor to set the value of 'area'. Instead, calculate the area in a separate method after the object has been constructed, or directly assign the calculated area within the constructor if necessary.